exampleCourse/questions/demo/annotated/engCircuit/server.py

```diff
@@ -1,40 +1,108 @@
 import random
 
+import prairielearn as pl
+import schemdraw.elements as elm
+from schemdraw import Drawing
 
-def generate(data):
-    ask = ["equivalent resistance $R_T$", "current from the power supply $I_T$"]
-    which = random.choice([0, 1])
-    data["params"]["ask"] = ask[which]
 
-    label = ["$R_T$", "$I_T$"]
-    data["params"]["lab"] = label[which]
+def file(data):
+    if data["filename"] != "figure.svg":
+        return
+
+    drawing = Drawing()
+    params_dict = data["params"]
+
+    match params_dict["whichfig"]:
+        case 0:
+            # variant: Resistors in series
+            drawing.push()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().right()
+            drawing += elm.Line().down()
+            drawing += elm.Resistor().left().label(params_dict["R3_label"])
+            drawing += elm.Resistor().left().label(params_dict["R2_label"])
+            drawing += elm.Resistor().left().label(params_dict["R1_label"])
+            drawing.pop()
+            drawing += (
+                elm.BatteryCell().down().label(["$+$", params_dict["Vt_label"], "$-$"])
+            )
+
+        case 1:
+            # variant: Resistors in parallel
+            drawing.push()
+            drawing += elm.Line()
+            drawing.push()
+            drawing += elm.Resistor().down().label(params_dict["R1_label"])
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing.push()
+            drawing += elm.Resistor().down().label(params_dict["R2_label"])
+            drawing.pop()
+            drawing += elm.Line().right()
+            drawing += elm.Resistor().down().label(params_dict["R3_label"])
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing += elm.Line().left()
+            drawing.pop()
+            drawing += (
+                elm.BatteryCell().down().label(["$+$", params_dict["Vt_label"], "$-$"])
+            )
 
-    unit = ["$\\Omega$", "A"]
-    data["params"]["unit"] = unit[which]
+    return drawing.get_imagedata()
 
-    Vt = random.randint(100, 200)
-    data["params"]["Vt"] = Vt
 
-    R1 = random.choice(list(range(20, 180, 10)))
-    data["params"]["R1"] = R1
+def generate(data):
+    ureg = pl.get_unit_registry()
+    params_dict = data["params"]
+
+    # Randomly choose Vt, R1, R2, R3 with appropriate units
+    Vt = random.randint(100, 200) * ureg.volt
+    R1 = random.choice(list(range(20, 180, 10))) * ureg.ohm
+    R2 = random.choice(list(range(20, 180, 20))) * ureg.ohm
+    R3 = random.choice(list(range(20, 100, 5))) * ureg.ohm
```

Comment:
`list` is not necessary here.
```suggestion
    R1 = random.choice(range(20, 180, 10)) * ureg.ohm
    R2 = random.choice(range(20, 180, 20)) * ureg.ohm
    R3 = random.choice(range(20, 100, 5)) * ureg.ohm
```

Reply:
Note to self: Use [this](https://docs.python.org/3/library/random.html#random.randrange).

Reply:
Done here https://github.com/PrairieLearn/PrairieLearn/pull/8179/commits/85e0c6b9be23c804969d39dee155703ecbaaa412